x/claim/keeper/ibc_callbacks.go

```diff
@@ -0,0 +1,96 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tharsis/evmos/x/claim/types"
+
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/cosmos/ibc-go/v3/modules/core/exported"
+)
+
+// OnRecvPacket performs an IBC callback.
+func (k Keeper) OnRecvPacket(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	ack exported.Acknowledgement,
+) exported.Acknowledgement {
+	params := k.GetParams(ctx)
+
+	if !ack.Success() || !params.IsClaimActive(ctx.BlockTime()) {
+		return ack
+	}
+
+	var data transfertypes.FungibleTokenPacketData
+	if err := transfertypes.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
+		err = sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
+		return channeltypes.NewErrorAcknowledgement(err.Error())
+	}
```

Comment:
:bug: 

```suggestion
	if err := transfertypes.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
		err = sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data")
		return channeltypes.NewErrorAcknowledgement(err.Error())
	}
```
JSON unmarshal error strings are non-determinstic (they use maps). If included in the acknowledgement, it'd cause an app hash mismatch